### ISSUE TITLE
Error in nn.MaxPool3d() ?

### ISSUE THREAD

**Issue by kobenaxie** [pytorch/pytorch#6661]
- OS:centos 6
- PyTorch version:0.3.1
- How you installed PyTorch (conda, pip, source):pip
- Python version:3.6.4
- CUDA/cuDNN version:8.0/6
-------------------------------------------------------
when specify the `padding=1` in `MaxPool3d()`, It will raise error.
 `In [1]: import torch

In [2]: input = torch.autograd.Variable(torch.randn(1, 75, 50, 100, 3), requires_grad=False)

In [3]: m = torch.nn.MaxPool3d(kernel_size=(1,2,2), stride=(1,2,2))

In [4]: out = m(input.permute(0, 4, 1, 2, 3))

In [5]: out.shape
Out[5]: torch.Size([1, 3, 75, 25, 50])

In [6]: m = torch.nn.MaxPool3d(kernel_size=(1,2,2), stride=(1,2,2), padding=1)
In [7]: out = m(input.permute(0, 4, 1, 2, 3))`
will raise error like this
*** glibc detected *** /home/xiejiliang/miniconda3/bin/python: free(): invalid pointer: 0x00007fdf21090d40 ***
======= Backtrace: =========
/lib64/libc.so.6(+0x318ae75366)[0x7fdf78e2f366]
/home/xiejiliang/miniconda3/lib/python3.6/site-packages/torch/_thnn/_THNN.cpython-36m-x86_64-linux-gnu.so(+0x7dd47)[0x7fdeefc07d47]
/home/xiejiliang/miniconda3/bin/python(_PyCFunction_FastCallDict+0x91)[0x7fdf79699f11]
/home/xiejiliang/miniconda3/bin/python(+0x19cbec)[0x7fdf79727bec]
/home/xiejiliang/miniconda3/bin/python(_PyEval_EvalFrameDefault+0x30a)[0x7fdf7974c19a]
/home/xiejiliang/miniconda3/bin/python(PyEval_EvalCodeEx+0x329)[0x7fdf79722529]
/home/xiejiliang/miniconda3/bin/python(+0x1983b9)[0x7fdf797233b9]
/home/xiejiliang/miniconda3/bin/python(PyObject_Call+0x3e)[0x7fdf79699dde]
/home/xiejiliang/miniconda3/lib/python3.6/site-packages/torch/_C.cpython-36m-x86_64-linux-gnu.so(_Z17THPFunction_applyP7_objectS0_+0x271)[0x7fdf1fb98ac1]
/home/xiejiliang/miniconda3/bin/python(_PyCFunction_FastCallDict+0x91)[0x7fdf79699f11]
/home/xiejiliang/miniconda3/bin/python(+0x19cbec)[0x7fdf79727bec]
/home/xiejiliang/miniconda3/bin/python(_PyEval_EvalFrameDefault+0x30a)[0x7fdf7974c19a]
/home/xiejiliang/miniconda3/bin/python(+0x1959a6)[0x7fdf797209a6]
/home/xiejiliang/miniconda3/bin/python(+0x196a11)[0x7fdf79721a11]
/home/xiejiliang/miniconda3/bin/python(+0x19ccc5)[0x7fdf79727cc5]
/home/xiejiliang/miniconda3/bin/python(_PyEval_EvalFrameDefault+0x30a)[0x7fdf7974c19a]
/home/xiejiliang/miniconda3/bin/python(_PyFunction_FastCallDict+0x11b)[0x7fdf79721e4b]
/home/xiejiliang/miniconda3/bin/python(_PyObject_FastCallDict+0x26f)[0x7fdf7969a39f]
/home/xiejiliang/miniconda3/bin/python(_PyObject_Call_Prepend+0x63)[0x7fdf7969eff3]
/home/xiejiliang/miniconda3/bin/python(PyObject_Call+0x3e)[0x7fdf79699dde]
/home/xiejiliang/miniconda3/bin/python(_PyEval_EvalFrameDefault+0x1b04)[0x7fdf7974d994]
/home/xiejiliang/miniconda3/bin/python(+0x1959a6)[0x7fdf797209a6]
/home/xiejiliang/miniconda3/bin/python(_PyFunction_FastCallDict+0x1be)[0x7fdf79721eee]
/home/xiejiliang/miniconda3/bin/python(_PyObject_FastCallDict+0x26f)[0x7fdf7969a39f]
/home/xiejiliang/miniconda3/bin/python(_PyObject_Call_Prepend+0x63)[0x7fdf7969eff3]
/home/xiejiliang/miniconda3/bin/python(PyObject_Call+0x3e)[0x7fdf79699dde]
/home/xiejiliang/miniconda3/bin/python(+0x16a901)[0x7fdf796f5901]
/home/xiejiliang/miniconda3/bin/python(_PyObject_FastCallDict+0x8b)[0x7fdf7969a1bb]
/home/xiejiliang/miniconda3/bin/python(+0x19cd3e)[0x7fdf79727d3e]
/home/xiejiliang/miniconda3/bin/python(_PyEval_EvalFrameDefault+0x30a)[0x7fdf7974c19a]
/home/xiejiliang/miniconda3/bin/python(PyEval_EvalCodeEx+0x329)[0x7fdf79722529]
/home/xiejiliang/miniconda3/bin/python(PyEval_EvalCode+0x1c)[0x7fdf797232cc]
/home/xiejiliang/miniconda3/bin/python(+0x1beb9b)[0x7fdf79749b9b]
/home/xiejiliang/miniconda3/bin/python(_PyCFunction_FastCallDict+0x91)[0x7fdf79699f11]
/home/xiejiliang/miniconda3/bin/python(+0x19cbec)[0x7fdf79727bec]
/home/xiejiliang/miniconda3/bin/python(_PyEval_EvalFrameDefault+0x30a)[0x7fdf7974c19a]
/home/xiejiliang/miniconda3/bin/python(+0x1959a6)[0x7fdf797209a6]
/home/xiejiliang/miniconda3/bin/python(+0x196a11)[0x7fdf79721a11]
/home/xiejiliang/miniconda3/bin/python(+0x19ccc5)[0x7fdf79727cc5]
/home/xiejiliang/miniconda3/bin/python(_PyEval_EvalFrameDefault+0x30a)[0x7fdf7974c19a]
/home/xiejiliang/miniconda3/bin/python(+0x1959a6)[0x7fdf797209a6]
/home/xiejiliang/miniconda3/bin/python(+0x196a11)[0x7fdf79721a11]
/home/xiejiliang/miniconda3/bin/python(+0x19ccc5)[0x7fdf79727cc5]
/home/xiejiliang/miniconda3/bin/python(_PyEval_EvalFrameDefault+0x1021)[0x7fdf7974ceb1]
/home/xiejiliang/miniconda3/bin/python(+0x195dfe)[0x7fdf79720dfe]
/home/xiejiliang/miniconda3/bin/python(+0x196a11)[0x7fdf79721a11]
/home/xiejiliang/miniconda3/bin/python(+0x19ccc5)[0x7fdf79727cc5]
/home/xiejiliang/miniconda3/bin/python(_PyEval_EvalFrameDefault+0x1021)[0x7fdf7974ceb1]
/home/xiejiliang/miniconda3/bin/python(+0x1959a6)[0x7fdf797209a6]
/home/xiejiliang/miniconda3/bin/python(+0x196a11)[0x7fdf79721a11]
/home/xiejiliang/miniconda3/bin/python(+0x19ccc5)[0x7fdf79727cc5]
/home/xiejiliang/miniconda3/bin/python(_PyEval_EvalFrameDefault+0x30a)[0x7fdf7974c19a]
/home/xiejiliang/miniconda3/bin/python(+0x1959a6)[0x7fdf797209a6]
/home/xiejiliang/miniconda3/bin/python(+0x196a11)[0x7fdf79721a11]
/home/xiejiliang/miniconda3/bin/python(+0x19ccc5)[0x7fdf79727cc5]
/home/xiejiliang/miniconda3/bin/python(_PyEval_EvalFrameDefault+0x30a)[0x7fdf7974c19a]
/home/xiejiliang/miniconda3/bin/python(+0x1967db)[0x7fdf797217db]
/home/xiejiliang/miniconda3/bin/python(+0x19ccc5)[0x7fdf79727cc5]
/home/xiejiliang/miniconda3/bin/python(_PyEval_EvalFrameDefault+0x30a)[0x7fdf7974c19a]
/home/xiejiliang/miniconda3/bin/python(+0x1959a6)[0x7fdf797209a6]
/home/xiejiliang/miniconda3/bin/python(_PyFunction_FastCallDict+0x3d8)[0x7fdf79722108]
/home/xiejiliang/miniconda3/bin/python(_PyObject_FastCallDict+0x26f)[0x7fdf7969a39f]
/home/xiejiliang/miniconda3/bin/python(_PyObject_Call_Prepend+0x63)[0x7fdf7969eff3]
======= Memory map: ========
7fdeefb8a000-7fdeefca4000 r-xp 00000000 08:08 21135602                   /home/xiejiliang/miniconda3/lib/python3.6/site-packages/torch/_thnn/_THNN.cpython-36m-x86_64-linux-gnu.so
7fdeefca4000-7fdeefea4000 ---p 0011a000 08:08 21135602                   /home/xiejiliang/miniconda3/lib/python3.6/site-packages/torch/_thnn/_THNN.cpython-36m-x86_64-linux-gnu.so
7fdeefea4000-7fdeefead000 rw-p 0011a000 08:08 21135602                   /home/xiejiliang/miniconda3/lib/python3.6/site-packages/torch/_thnn/_THNN.cpython-36m-x86_64-linux-gnu.so
7fdeefead000-7fdef034d000 rw-p 00000000 00:00 0 
7fdef034d000-7fdef034e000 ---p 00000000 00:00 0 
7fdef034e000-7fdef128e000 rw-p 00000000 00:00 0 
7fdef128e000-7fdef3b7f000 r-xp 00000000 08:08 21037070                   /home/xiejiliang/miniconda3/lib/python3.6/site-packages/torch/lib/libcusparse-94011b8d.so.8.0.61
7fdef3b7f000-7fdef3d7f000 ---p 028f1000 08:08 21037070                   /home/xiejiliang/miniconda3/lib/python3.6/site-packages/torch/lib/libcusparse-94011b8d.so.8.0.61
7fdef3d7f000-7fdef3d98000 rw-p 028f1000 08:08 21037070                   /home/xiejiliang/miniconda3/lib/python3.6/site-packages/torch/lib/libcusparse-94011b8d.so.8.0.61
7fdef3d98000-7fdef3da9000 rw-p 00000000 00:00 0 
7fdef3da9000-7fdef3daf000 rw-p 0290b000 08:08 21037070                   /home/xiejiliang/miniconda3/lib/python3.6/site-packages/torch/lib/libcusparse-94011b8d.so.8.0.61
7fdef3daf000-7fdef624a000 r-xp 00000000 08:08 21037071                   /home/xiejiliang/miniconda3/lib/python3.6/site-packages/torch/lib/libcurand-3d68c345.so.8.0.61
7fdef624a000-7fdef644a000 ---p 0249b000 08:08 21037071                   /home/xiejiliang/miniconda3/lib/python3.6/site-packages/torch/lib/libcurand-3d68c345.so.8.0.61
7fdef644a000-7fdef781b000 rw-p 0249b000 08:08 21037071                   /home/xiejiliang/miniconda3/lib/python3.6/site-packages/torch/lib/libcurand-3d68c345.so.8.0.61
7fdef781b000-7fdef7d25000 rw-p 00000000 00:00 0 
7fdef7d25000-7fdef7d26000 rw-p 0386d000 08:08 21037071                   /home/xiejiliang/miniconda3/lib/python3.6/site-packages/torch/lib/libcurand-3d68c345.so.8.0.61
7fdef7d26000-7fdefab3e000 r-xp 00000000 08:08 21037072                   /home/xiejiliang/miniconda3/lib/python3.6/site-packages/torch/lib/libcublas-e78c880d.so.8.0.88
7fdefab3e000-7fdefad3e000 ---p 02e18000 08:08 21037072                   /home/xiejiliang/miniconda3/lib/python3.6/site-packages/torch/lib/libcublas-e78c880d.so.8.0.88
7fdefad3e000-7fdefad5c000 rw-p 02e18000 08:08 21037072                   /home/xiejiliang/miniconda3/lib/python3.6/site-packages/torch/lib/libcublas-e78c880d.so.8.0.88
7fdefad5c000-7fdefad6b000 rw-p 00000000 00:00 0 
7fdefad6b000-7fdefad6f000 rw-p 02e36000 08:08 21037072                   /home/xiejiliang/miniconda3/lib/python3.6/site-packages/torch/lib/libcublas-e78c880d.so.8.0.88
7fdefad6f000-7fdefad84000 r-xp 00000000 08:08 21037069                   /home/xiejiliang/miniconda3/lib/python3.6/site-packages/torch/lib/libgomp-ae56ecdc.so.1
7fdefad84000-7fdefaf83000 ---p 00015000 08:08 21037069                   /home/xiejiliang/miniconda3/lib/python3.6/site-packages/torch/lib/libgomp-ae56ecdc.so.1
7fdefaf83000-7fdefaf86000 rw-p 00014000 08:08 21037069                   /home/xiejiliang/miniconda3/lib/python3.6/site-packages/torch/lib/libgomp-ae56ecdc.so.1
7fdefaf86000-7fdefb071000 r-xp 00000000 08:02 2784                       /usr/local/gcc-4.8.5/lib64/libstdc++.so.6
7fdefb071000-7fdefb270000 ---p 000eb000 08:02 2784                       /usr/local/gcc-4.8.5/lib64/libstdc++.so.6
7fdefb270000-7fdefb278000 r--p 000ea000 08:02 2784                       /usr/local/gcc-4.8.5/lib64/libstdc++.so.6
7fdefb278000-7fdefb27a000 rw-p 000f2000 08:02 2784                       /usr/local/gcc-4.8.5/lib64/libstdc++.so.6
7fdefb27a000-7fdefb28f000 rw-p 00000000 00:00 0 
7fdefb28f000-7fdefb9c5000 r-xp 00000000 08:08 21037061                   /home/xiejiliang/miniconda3/lib/python3.6/site-packages/torch/lib/libnccl.so.1
7fdefb9c5000-7fdefbbc5000 ---p 00736000 08:08 21037061                   /home/xiejiliang/miniconda3/lib/python3.6/site-packages/torch/lib/libnccl.so.1
7fdefbbc5000-7fdefbbc6000 rw-p 00736000 08:08 21037061                   /home/xiejiliang/miniconda3/lib/python3.6/site-packages/torch/lib/libnccl.so.1
7fdefbbc6000-7fdefbbc8000 rw-p 00742000 08:08 21037061                   /home/xiejiliang/miniconda3/lib/python3.6/site-packages/torch/lib/libnccl.so.1
7fdefbbc8000-7fdefecc3000 r-xp 00000000 08:08 21037074                   /home/xiejiliang/miniconda3/lib/python3.6/site-packages/torch/lib/libTHCUNN.so.1
7fdefecc3000-7fdefeec2000 ---p 030fb000 08:08 21037074                   /home/xiejiliang/miniconda3/lib/python3.6/site-packages/torch/lib/libTHCUNN.so.1
7fdefeec2000-7fdefeed8000 rw-p 030fa000 08:08 21037074                   /home/xiejiliang/miniconda3/lib/python3.6/site-packages/torch/lib/libTHCUNN.so.1
7fdefeed8000-7fdefef1f000 rw-p 00000000 00:00 0 
7fdefef1f000-7fdeff20c000 rw-p 034a5000 08:08 21037074                   /home/xiejiliang/miniconda3/lib/python3.6/site-packages/torch/lib/libTHCUNN.so.1
7fdeff20c000-7fdeff396000 r-xp 00000000 08:08 21037058                   /home/xiejiliang/miniconda3/lib/python3.6/site-packages/torch/lib/libTHCS.so.1已放弃

**Comment by zou3519** [comment]
I'm getting the following error on master:
![image](https://user-images.githubusercontent.com/5652049/38878548-2e5abf98-422f-11e8-92b9-aab7c037f01e.png)

**Comment by apaszke** [comment]
It's the classic invalid free on error. @soumith Was this supposed to be fixed by now?
